airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableCell.tsx

```diff
@@ -22,67 +23,67 @@ const sizeMap: Record<Sizes, string> = {
   large: styles.large,
 };
 
-const TooltipText: React.FC<{ textNodes: Element[] }> = ({ textNodes }) => {
-  if (!textNodes.length) {
-    return null;
-  }
-  const text = textNodes.map((t) => decodeURIComponent(t.innerHTML)).join(" | ");
-  // This is not a safe use, and need to be removed still.
-  // https://github.com/airbytehq/airbyte/issues/22196
-  // eslint-disable-next-line react/no-danger
-  return <div dangerouslySetInnerHTML={{ __html: text }} />;
-};
-
 export const CatalogTreeTableCell: React.FC<React.PropsWithChildren<CatalogTreeTableCellProps>> = ({
   size = "medium",
   withTooltip,
   className,
   children,
 }) => {
   const [tooltipDisabled, setTooltipDisabled] = useState(true);
-  const [textNodes, setTextNodes] = useState<Element[]>([]);
-  const cell = useRef<HTMLDivElement | null>(null);
+  const cellContent = useRef<HTMLSpanElement | null>(null);
 
-  const { width: windowWidth } = useWindowSize();
+  const { inView, ref: inViewRef } = useInView({ delay: 500 });
 
-  const getTextNodes = useCallback(() => {
-    if (withTooltip && cell.current) {
-      setTextNodes(Array.from(cell.current.querySelectorAll(`[data-type="text"]`)));
+  useEffect(() => {
+    if (!inView || !withTooltip) {
+      // Only handle resize events on the window for this cell (to determine if the tooltip)
+      // needs to be shown if the cell is actually in view
+      return;
     }
-  }, [withTooltip]);
 
-  useEffect(() => {
-    // windowWidth is only here so this functionality changes based on window width
-    if (textNodes.length && windowWidth) {
-      const [scrollWidths, clientWidths] = textNodes.reduce(
-        ([scrollWidths, clientWidths], textNode) => {
-          if (textNode) {
-            scrollWidths += textNode.scrollWidth;
-            clientWidths += textNode.clientWidth;
-          }
-          return [scrollWidths, clientWidths];
-        },
-        [0, 0]
+    // Calculate based on any potentially truncated `<Text>` element inside this cell, whether
+    // the tooltip should show.
+    const calculateTooltipVisible = () => {
+      const hasEllipsisedElement = Array.from(cellContent.current?.querySelectorAll(`[data-type="text"]`) ?? []).some(
```

Comment:
This was the easiest way I had to gather multiple text nodes together. Specifically around the sync mode which has since been removed. I think this could be simplified now.